langchain/src/retrievers/tests/vectara.int.test.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-process-env */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { test, expect } from "@jest/globals";
+
+import { VectaraRetriever } from "../vectara.js";
+import { VectaraStore } from "../../vectorstores/vectara.js";
+
+/**
+ * Steps to run this test:
+ * 1. Create a Vectara account at https://vectara.com/
+ * 2. Create a corpus
+ * 3. Upload a specific document to the corpus (e.g. the US constitution)
+ * 4. Set the appropriate environment variables (VECTARA_CUSTOMER_ID, VECTARA_CORPUS_ID, VECTARA_API_KEY)
+ * 5. Run the test
+ */
+
+test("VectaraRetriever (with filter)", async () => {
+  const store = new VectaraStore({
+    customer_id: process.env.VECTARA_CUSTOMER_ID ? parseInt(process.env.VECTARA_CUSTOMER_ID) : 0,
+    corpus_id: process.env.VECTARA_CORPUS_ID ? parseInt(process.env.VECTARA_CORPUS_ID) : 0,
+    api_key: process.env.VECTARA_API_KEY || "",
+  });
+  const retriever = new VectaraRetriever(store);
+
+  // If you've uploaded the US constituion, this will return the relevant section
+  const docs = await retriever.getRelevantDocuments("Article 1 Section 7");
+
+  expect(docs.length).toBeGreaterThan(0);
+  // This checks that the document contains the relevant section
+  expect(docs[0].pageContent).toContain('All Bills for raising Revenue shall originate in the House of Representatives');
+
+  console.log(docs);
+});
+
+test("VectaraRetriever (without filter)", async () => {
+  const store = new VectaraStore({
+    customer_id: process.env.VECTARA_CUSTOMER_ID ? parseInt(process.env.VECTARA_CUSTOMER_ID) : 0,
+    corpus_id: process.env.VECTARA_CORPUS_ID ? parseInt(process.env.VECTARA_CORPUS_ID) : 0,
+    api_key: process.env.VECTARA_API_KEY || "",
+  });
+  const retriever = new VectaraRetriever(store);
+
+  const docs = await retriever.getRelevantDocuments("hello");
+
+  expect(docs.length).toBeGreaterThan(0);
```

Comment:
Is it worth asserting against `docs[0].pageContent` like we do in the first test?

Reply:
The difference between the first and second test is the assert (related to conversation above).

Reply:
some of my perspective on this:
1. Testing should be as meaningful/useful as you can make it.
2. I think what CJ is saying is that the 2nd test (the one testing that you get more than 0 documents matching "hello") is not super useful (it tests for whether any doc is returned, but that's sort of included in the first test). 
3. My recommendation would be to try and test for text that is meaningful for the ingested documents. So if we've uploaded the US constitution, then let's figure out a good query (something better than "hello") and test that the right response is provided.
4. What I would expect for with-filter vs w/o filter is the following: w/o filter is we do a query and check that some docs are returned and they are the correct content. With filter would use Vectara's meta-data fields as a filter. So you would for example ingest a document but make sure that the corpus has meta-data, and then filter by that meta-data. @3eif - note that you have to specify that meta-data when creating the corpus for this to work.
5. It would be also super helpful to add some documentation that will allow others to really use the test,by describing the steps used to "ingest" the docs used in the test.